File: Subathra_Assessment7/UserRegistrationAndLogin/src/test/java/com/Pages/LoginPage.java
package com.Pages;

import java.time.Duration;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class LoginPage extends BasePage {
	
	BasePage basePage;
	
	WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(15));
	
	
	public LoginPage(WebDriver driver) {
		super(driver);
	}
	
	@FindBy (xpath="//input[@data-qa='login-email']")
	public WebElement email;
	
	@FindBy (xpath="//input[@name=\"password\"]")
	public WebElement password;
	
	@FindBy (xpath="//button[text()='Login']")
	public WebElement loginButton;
	
	@FindBy (xpath="//a[text()=' Logout']")
	public WebElement logout;
	
	@FindBy (xpath="//p[text()='Your email or password is incorrect!']")
	public WebElement invalid;
	
	
	
	public void login(String log_email,String log_pass) {
		
		basePage = new BasePage(driver);
		
		basePage.clickSignupOrLogin();
	
		wait.until(ExpectedConditions.visibilityOf(email));
		
		email.sendKeys(log_email);
		password.sendKeys(log_pass);
		loginButton.click();
	
	}
	
	public String getText() {
		
		wait.until(ExpectedConditions.visibilityOf(logout));
		return logout.getText();
	}
	
	
	public String getInvalidMessage() {
		
		return invalid.getText();
	}
	
	

}

File: BDD_Cucumber_Assessment/src/test/pages/loginPage.ts
import { Page, Locator } from 'playwright'


export class LoginPage {

    readonly page: Page;
    readonly email: Locator;
    readonly password: Locator;
    readonly loginButton: Locator;

    constructor(page: Page) {
        this.page = page;
        this.email = this.page.getByLabel("Email:");
        this.password = this.page.getByLabel("Password:");
        this.loginButton = this.page.locator("//input[@value='Log in']");
    }

    async enterLoginDetails(email: string, password:string) {
        await this.email.fill(email);
        await this.password.fill(password);
    }

    async ClickLoginButton() {
        await this.loginButton.click();
    }
}